query.go

```diff
@@ -6,10 +6,6 @@ import (
 
 // Query represents an active query.
 type Query interface {
```

Comment:
I would have expected the Ready method to need to change to be a `<-chan Result` instead of a `<-chan map[string]Result`. The difference being that now the Ready channel can produce more than one set of results. In fact the name Ready is inaccurate now since its not about the query being ready but just a mechanism to deliver results.

If this should be a follow up PR that is fine, lets create an issue for it.